ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java

```diff
@@ -0,0 +1,1240 @@
+          node.getName(),
+          node.isStdOut,
+          properties);
+    } else {
+      return new Table(node.getName(),
+          node.isStdOut,
+          properties);
+    }
+
+  }
+
+  protected Node visitValues(Values node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new Values(node.getLocation().get(),
+          node.getRows().stream()
+              .map(row -> (Expression) process(row, context))
+              .collect(Collectors.toList())
+      );
+    } else {
+      return new Values(node.getRows().stream()
+          .map(row -> (Expression) process(row, context))
+          .collect(Collectors.toList())
+      );
+    }
+
+  }
+
+  protected Node visitStruct(Struct node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new Struct(node.getLocation().get(),
+          node.getItems());
+    } else {
+      return new Struct(node.getItems());
+    }
+
+  }
+
+  protected Node visitTableSubquery(TableSubquery node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new TableSubquery(node.getLocation().get(),
+          (Query) process(node.getQuery(), context)
+      );
+    } else {
+      return new TableSubquery((Query) process(node.getQuery(), context)
+      );
+    }
+
+  }
+
+  protected Node visitAliasedRelation(AliasedRelation node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new AliasedRelation(node.getLocation().get(),
+          (Relation) process(node.getRelation(), context),
+          node.getAlias(),
+          node.getColumnNames());
+    } else {
+      return new AliasedRelation((Relation) process(node.getRelation(), context),
+          node.getAlias(),
+          node.getColumnNames());
+    }
+
+  }
+
+  protected Node visitSampledRelation(SampledRelation node, Object context) {
+
+    Optional<List<Expression>> columnsToStratifyOn = node.getColumnsToStratifyOn().isPresent()
+        ? Optional.ofNullable(
+        node.getColumnsToStratifyOn().get()
+            .stream()
+            .map(expression -> (Expression) process(expression, context))
+            .collect(Collectors.toList()))
+        : Optional.empty();
+    if (node.getLocation().isPresent()) {
+      return new SampledRelation(node.getLocation().get(),
+          (Relation) process(node.getRelation(), context),
+          node.getType(),
+          (Expression) process(node.getSamplePercentage(), context),
+          node.isRescaled(),
+          columnsToStratifyOn
+      );
+    } else {
+      return new SampledRelation((Relation) process(node.getRelation(), context),
+          node.getType(),
+          (Expression) process(node.getSamplePercentage(), context),
+          node.isRescaled(),
+          columnsToStratifyOn
+      );
+    }
+
+  }
+
+  protected Node visitJoin(Join node, Object context) {
+    //TODO: Will have to look into Criteria later (includes Expression)
```

Comment:
This seems fine for now since the join key can't be a nested field in a struct.

Reply:
Yes, no change on this at the moment.